aptos-move/framework/aptos-experimental/sources/trading/market/market.move

```diff
@@ -727,28 +771,53 @@ module aptos_experimental::market {
                 0, // 0 because order was never placed
                 vector[],
                 is_bid,
-                true, // is_taker
+                is_taker_order, // is_taker
                 OrderCancellationReason::PositionUpdateViolation,
                 std::string::utf8(b"Position Update violation"),
                 callbacks
             );
         };
 
-        if (emit_taker_order_open) {
-            self.emit_event_for_order(
-                order_id,
-                client_order_id,
+        if (client_order_id.is_some()) {
+            if (self.order_book.client_order_id_exists(user_addr, client_order_id.destroy_some())) {
```

Comment:
Can you add a comment here describing when this scenario is possible?

Reply:
I think we should explain at the top when this should be used,  and maybe rename from precancelation.

Flow is:
User submits a “place_order” txn. 

then shortly thereafter, before place order is committed,  user changes their mind,  and wants to cancel their order. It submits a precancel, such that irrespective of how blockchain orders those two transactions,  unless order gets matched before cancel is committed,  order is canceled